embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -683,12 +693,37 @@ public boolean accept(File file) {
     protected void addDriverJarToClasspath(String glob)
     {
         // TODO match glob
-        PluginClassLoader loader = (PluginClassLoader) getClass().getClassLoader();
+        final ClassLoader loader = getClass().getClassLoader();
+        if (!(loader instanceof URLClassLoader)) {
+            throw new RuntimeException("Plugin is not loaded by URLClassLoader unexpectedly.");
+        }
+        if (!loader.getClass().getCanonicalName().equals("org.embulk.plugin.PluginClassLoader")) {
```

Comment:
`Class#getCanonicalName` may return `null`.
https://docs.oracle.com/javase/8/docs/api/java/lang/Class.html#getCanonicalName--

Can we use `Class#getName` instead?

Reply:
Ah, thanks for your good catch. `null.equals(...)` would result in unexpected failure there. (It should fail anyway eventually, but it should fail in an expected way.)

I at first thought that a "canonical" name would sound better to confirm a kind of equality, but reading the Q&A below, I understood that `getName()` would be better.

> * the **name** is the name that you'd use to dynamically load the class with, for example, a call to `Class.forName` with the default `ClassLoader`. Within the scope of a certain `ClassLoader`, **all classes have unique names**.
> * the **canonical name** is the name that would be used in an import statement. It might be useful during `toString` or logging operations. ... (snip) ... thus **canonical names do not uniquely identify classes within a `ClassLoader`**.
>
> https://stackoverflow.com/questions/15202997/what-is-the-difference-between-canonical-name-simple-name-and-class-name-in-jav

Then in c56539e9f170b4a58ae3748b5e4cecaee2b763ff, I :
* changed it to `getName()`, and
* reversed the condition like below so that it works even if `getName()` returns an unexpected thing evem in the future.

```
       if (!"org.embulk.plugin.PluginClassLoader".equals(loader.getClass().getName())) {
```

What do you think?

Reply:
LGTM!